src/java/htsjdk/samtools/SAMFileWriterImpl.java

```diff
@@ -171,9 +171,6 @@ private SAMRecordComparator makeComparator() {
     public void addAlignment(final SAMRecord alignment)
     {
         if (sortOrder.equals(SAMFileHeader.SortOrder.unsorted)) {
-            if (!header.getGroupOrder().equals(SAMFileHeader.GroupOrder.none)) {
-                throw new UnsupportedOperationException("GroupOrder " + header.getGroupOrder() + " is not supported");
-            }
```

Comment:
I suppose I would want to understand why this check was here in the first place, given no where else do we consider the group order, and this only is an issue when writing.  I suppose it is because we would pass the alignment to the `alignmentSorter` if we request a sort order (`SO`) but do nothing in the case of group order (`GO`).  In your case, you are adding support `SO=unsorted` and `GO=query`, so shouldn't the alignment be passed to an alignmentSorter based on your new comparator if `SO=unsorted` and `GO=query`?